rpmconf/Upgrade/zmupgrade.pm

```diff
@@ -407,6 +407,7 @@ sub upgrade {
       if ($schema_found) {
         $needMysqlUpgrade=1 if ($v eq "8.0.0_GA");
         $needMysqlUpgrade=1 if ($v eq "8.5.0_BETA1");
+        $needMysqlUpgrade=1 if ($v eq "8.7.10_GA");
```

Comment:
I think (if I understood the code correctly) this should work for 8.7.11 as well.

But you're right, I think this is a part which is prone to be forgotten in the future as well. My alternate suggestion would be to always run `mysql_upgrade` except on patch releases (`$needMysqlUpgrade = ($startMajor < $targetMajor) || ($startMinor < $targetMinor);`). I don't think/hope the MariaDB version will ever be changed between patch releases. And people already (should) plan longer downtimes for these updates.

Maybe additionally the user could choose to skip this like the `zmdbintegrityreport`.

Plus, there should be a `zmmysqlupgrade` wrapper which provides the proper command line parameters. Or maybe just a properly configured `my.cnf` so `mysql_upgrade` Just Works. The Release Notes should mention that this has to be done for installations which were upgraded from < 8.7.0 once.